src/KeyManagement.sol

```diff
@@ -56,14 +58,35 @@ abstract contract KeyManagement is IKeyManagement, BaseAuthorization {
     /// @inheritdoc IKeyManagement
     function getKey(bytes32 keyHash) public view returns (Key memory) {
         if (keyHash.isRootKey()) return KeyLib.toRootKey();
-        if (keyHashes.contains(keyHash)) return abi.decode(keyStorage[keyHash], (Key));
+        if (isRegistered(keyHash)) return abi.decode(keyStorage[keyHash], (Key));
         revert KeyDoesNotExist();
     }
 
     /// @inheritdoc IKeyManagement
     function getKeySettings(bytes32 keyHash) public view returns (Settings) {
         if (keyHash.isRootKey()) return SettingsLib.ROOT_KEY_SETTINGS;
-        if (keyHashes.contains(keyHash)) return keySettings[keyHash];
+        if (isRegistered(keyHash)) return keySettings[keyHash];
         revert KeyDoesNotExist();
     }
+
+    /// @inheritdoc IKeyManagement
+    function isRegistered(bytes32 keyHash) public view returns (bool) {
+        return keyHashes.contains(keyHash);
+    }
+
+    /// @inheritdoc IKeyManagement
+    function isAuthorizedAdmin(bytes32 keyHash) public view returns (bool) {
+        if (!isRegistered(keyHash)) return false;
+        if (keyHash.isRootKey()) return true;
+        Settings settings = keySettings[keyHash];
+        if (settings.isAdmin()) return true;
+        return false;
+    }
+
+    /// @inheritdoc BaseAuthorization
+    /// @dev This function is a wrapper around isAuthorizedAdmin(bytes32) to abstract the keyHash/key implememtation from the BaseAuthorization contract.
+    function isAuthorizedAdmin(address caller) public view override(BaseAuthorization) returns (bool) {
+        Key memory key = Key({keyType: KeyType.Secp256k1, publicKey: abi.encode(caller)});
```

Comment:
Would add a note how it's fine to set keyType to k1 here because these require originating transactions